packages/backend/src/index.ts

```diff
@@ -84,86 +84,28 @@ function makeCreateEnv(config: Config, pluginProvider: BackendPluginProvider) {
   };
 }
 
-type AddPluginBase = {
-  isOptional?: boolean;
+async function addPlugin(args: {
   plugin: string;
   apiRouter: Router;
   createEnv: ReturnType<typeof makeCreateEnv>;
   router: (env: PluginEnvironment) => Promise<Router>;
-  options?: { path?: string };
-};
-
-type AddPlugin = {
-  isOptional?: false;
-} & AddPluginBase;
-
-type OptionalPluginOptions = {
-  key?: string;
-  path?: string;
-};
-
-type AddOptionalPlugin = {
-  isOptional: true;
-  config: Config;
-  options?: OptionalPluginOptions;
-} & AddPluginBase;
+}): Promise<void> {
+  const { plugin, apiRouter, createEnv, router } = args;
 
-const OPTIONAL_DYNAMIC_PLUGINS: { [key: string]: OptionalPluginOptions } = {
-  techdocs: {},
-  argocd: {},
-  sonarqube: {},
-  kubernetes: {},
-  'azure-devops': { key: 'enabled.azureDevOps' },
-  jenkins: {},
-  ocm: {},
-  gitlab: {},
-} as const satisfies { [key: string]: OptionalPluginOptions };
-
-async function addPlugin(args: AddPlugin | AddOptionalPlugin): Promise<void> {
-  const { isOptional, plugin, apiRouter, createEnv, router, options } = args;
-
-  const isPluginEnabled =
-    !isOptional ||
-    args.config.getOptionalBoolean(options?.key ?? `enabled.${plugin}`) ||
-    false;
-  if (isPluginEnabled) {
-    const pluginEnv: PluginEnvironment = useHotMemoize(module, () =>
-      createEnv(plugin),
-    );
-    apiRouter.use(options?.path ?? `/${plugin}`, await router(pluginEnv));
-    console.log(`Using backend plugin ${plugin}...`);
-  } else if (isOptional) {
-    console.log(`Backend plugin ${plugin} is disabled`);
-  }
+  const pluginEnv: PluginEnvironment = useHotMemoize(module, () =>
+    createEnv(plugin),
+  );
+  apiRouter.use(`/${plugin}`, await router(pluginEnv));
 }
 
-type AddRouterBase = {
-  isOptional?: boolean;
-  name: string;
+async function addRouter(args: {
   service: ServiceBuilder;
   root: string;
   router: RequestHandler | Router;
-};
+}): Promise<void> {
+  const { service, root, router } = args;
 
-type AddRouterOptional = {
-  isOptional: true;
-  config: Config;
-} & AddRouterBase;
-
-type AddRouter = {
-  isOptional?: false;
-} & AddRouterBase;
-
-async function addRouter(args: AddRouter | AddRouterOptional): Promise<void> {
-  const { isOptional, name, service, root, router } = args;
-
-  const isRouterEnabled =
-    !isOptional || args.config.getOptionalBoolean(`enabled.${name}`) || false;
-
-  if (isRouterEnabled) {
-    console.log(`Adding router ${name} to backend...`);
```

Comment:
Not a request, just a thought... I actually liked this log line. 😄

Reply:
I can add it back